### DISCUSSION TITLE
Set rank of claim as preferred and other claims to normal

### DISCUSSION THREAD

**Discussion by lubianat** [LeMyst/WikibaseIntegrator#566] [Q&A] [answered]
Hi! I am trying to add some new population statements for Brazilian municipalities. 
	
I am planning to adapt some pywikibot code I used for death counts: 
	
```
	        item = pywikibot.ItemPage(repo, row["item"])
        item.get()  # Fetch all page data, and cache it.

        # Change rank of current deaths to normal
        # (code based on https://gist.github.com/urbanecm/f0b2bb95514f322207d04908cc4d8143)
        itemData = item.get()
        death_claims = itemData["claims"].get("P1120", [])
        for claim in death_claims:
            if claim.rank != "preferred":
                continue
            claim.setRank("normal")
            item.editEntity(
                {"claims": [claim.toJSON()]}, summary="Change P1120 rank to normal"
            )

        # Add number of deaths

        ## Value --> death count

        deaths_claim = pywikibot.Claim(
            repo, u"P1120"
        )  # Adding number of deaths (P1120)
        deaths_claim.setTarget(
            pywikibot.WbQuantity(row["deaths"])
        )  # Set the target value in the local object.
        deaths_claim.setRank("preferred")
        item.addClaim(deaths_claim, summary=u"Adding death count from template")
```
I am trying to figure out if there is a way to do it using WikibaseIntegrator. Can anyone show me some pointers? Thanks! (@LeMyst or @dpriskorn )

**Comment by LeMyst** [comment, marked as answer]
Hello @lubianat 

I'm not sure if you want more info about setting a rank or updating population.
I did the operation for French Population earlier this year and updated this repo (mainly the _fastrun part):
https://github.com/LeMyst/wd-population

I think https://github.com/LeMyst/wd-population/blob/master/communes_fastrun.py can be a good start.

Line 98 I set all the rank to NORMAL (maybe do this only of claims who are actually preferred, to avoid setting a DEPRECATED to normal, I will fix this in my version 😅 )
Line 108 I set my new claim (in append or replace mode) with the rank PREFERRED, if it already exist, it will be updated.

If you have more question about the fastrun mode, you can ask them ;)

**Comment by lubianat** [comment]
oh, very good point about the deprecated ones!! I totally ignored it for the covid bot -- ops!  

Thank you so much for the reply! Maybe you can add a link to your handling somewhere in the repository, perhaps a "see also" section in the end of the README or something

**Comment by dpriskorn** [comment]
> oh, very good point about the deprecated ones!! I totally ignored it for the covid bot -- ops!  
> 
> Thank you so much for the reply! Maybe you can add a link to your handling somewhere in the repository, perhaps a "see also" section in the end of the README or something
> 

I usually comment and link directly above code inspired by others. This way others can easily see which parts I got from where. 😊

**Comment by LeMyst** [comment]
> oh, very good point about the deprecated ones!! I totally ignored it for the covid bot -- ops!
> 
> Thank you so much for the reply! Maybe you can add a link to your handling somewhere in the repository, perhaps a "see also" section in the end of the README or something

There is a "Other projects" section in the README:
https://github.com/LeMyst/WikibaseIntegrator#other-projects
